### PR TITLE
add Triple events, multiples strategy and changed CoincidenceEvent

### PR DESCRIPTION
**Pull request by KrisThielemans**
Adding triple coincidences. We're now using arrays of length 2 for CoincidenceEvents, such that it is the same for triples.
Also added a field for how multiples are handled by the scanner.

Also added documentation clarifying TOF conventions.

***BREAKS BACKWARDS COMPATIBILITY***

 - [x] The content of this Pull Request (the Contribution) is intentionally submitted for inclusion in the ETSI software (the Work) under the terms and conditions of the [Apache-2.0](https://www.apache.org/licenses/LICENSE-2.0) License.

### CODE REVIEW

**Comment by KrisThielemans** [comment]
@johnstairs how do we document enums in yardl? My current strategy for the `MultipleCoincidenceStrategy` doesn't seem to work.

File: model/ScannerInformation.yml [hunk 1]
```diff
@@ -9,12 +9,25 @@ Detector: !record
     y: float
     z: float
 
+# Type definition for how to encode how the scanner handles multiple coincidences
+MultipleCoincidenceStrategy: !enum
+  values:
+    - rejectMultiples          # multiples will be rejected
+    - storeMultiplesAsPairs    # multiples will be stored as a sequence of pairs, e.g. a triple leads to 3 pairs
```
Comment (johnstairs) [inline model/ScannerInformation.yml:16]: To document enum fields, you can add the comment above them:
```suggestion
# Type definition for how to encode how the scanner handles multiple coincidences
MultipleCoincidenceStrategy: !enum
  values:
    # multiples will be rejected
    - rejectMultiples

    # multiples will be stored as a sequence of pairs, e.g. a triple leads to 3 pairs
    - storeMultiplesAsPairs
```

**Comment by MaxTousss** [comment]
My knowledge about triple events is limited, so take the following with a grain of salt.

The only potential problem I have found with the proposed modifications is the definition of *MultipleCoincidenceStrategy*. 

1. Currently, it only defines rejection or conversion into coincidence pair. In other word, it does not have a value associated with the fact that the format also supports *tripleEvents*, where they are stored as triple. The intent might be to that one can easily check for the existence of *tripleEvents* but it feels prone to error.
2. Also, it feels like *MultipleCoincidenceStrategy* should refer to how multiple events are dealt with and not how they are represented/stored. If I use GATE as a reference, this name feels more appropriate for what GATE calls *Policy name* for multiple coincidences. Thus, my neophyte view on the matter would be to have *MultipleCoincidenceStrategy* defines two parameters: something akin to GATE's *Policy name* and something like *TripletContainer* or *TripletStructure*.
3. Lastly, my understanding of triplet events recovery is that you only want to work with triplets that has at least one of the three events with an energy of around 511 keV while also allowing small energy window for the other two. Thus, it feels like one would have two energy windows in that case? Unsure if it is possible in practice, however. If it is, we should be able to store that information in *Protocol* or something.

**Comment by KrisThielemans** [comment]
We have discussed this more in a previous meeting, after referring to table 8.2 of the [Gate-UsersGuideV7.2](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://gate.uca.fr/medias/fichier/gate-usersguidev7-2_1595341343803-pdf%3FID_FICHE%3D72603%26INLINE%3DFALSE&ved=2ahUKEwit7OjJ_5SGAxWNhP0HHcZLAPIQFnoECA4QAQ&usg=AOvVaw1LT07ZT9T0eGWOXveVJ9Fn).

We agree that "policy" is a better name than "strategy" and have adjusted it accordingly. Also added a little bit more clarification.

The topic of how scanners handle multiple single events is very complicated. For this PR, we just list 2 simple policies and an "other". We will aim to get information on common policies to be able to encode those.

@MaxTousss hopefully the new wording makes it clear that this only discusses how prompts are recorded. Similar issues with triples will be left for later. As an example, if the stream stores "triple coincidences", than it is expected that the "prompt coincidences" will be recorded with a policy of `rejectMultiples`. It is unclear to me if that is necessary to enforce this (so I wouldn't).



> 3\. Lastly, my understanding of triplet events recovery is that you only want to work with triplets that has at least one of the three events with an energy of around 511 keV while also allowing small energy window for the other two. Thus, it feels like one would have two energy windows in that case? Unsure if it is possible in practice, however. If it is, we should be able to store that information in _Protocol_ or something.

At present, (double) coincidences and triples store energy window information for every single (so, 2 energy windows for a double coincidence, etc). We believe this will address your concern.

**Comment by KrisThielemans** [comment]
> 1. The intent might be to that one can easily check for the existence of _tripleEvents_ but it feels prone to error.

That was indeed the intention, but we agree that this is difficult in practice. We might need to have booleans in the header `delayedsAreRecorded`, `triplesAreRecorded` etc.

**Comment by KrisThielemans** [comment]
This PR could easily be extended to quadruples, but we will leave that for later as well.

**Pull request merged**